compiler/circle-mpqsolver/src/bisection/VISQErrorApproximator.cpp

```diff
@@ -30,9 +31,28 @@ void VISQErrorApproximator::init(const std::string &visq_data_path)
   }
 }
 
-bool VISQErrorApproximator::init(std::istream &)
+bool VISQErrorApproximator::init(std::istream &visq_stream)
 {
-  // TODO
+  Json::Reader reader;
+  Json::Value completeJsonData;
+  if (!reader.parse(visq_stream, completeJsonData))
+  {
+    return false;
```

Comment:
How is the return value of this function? I think it is necessary to throw an exception if this function returns false.

Reply:
Ahhh. Sorry. Ok.

Reply:
done